fix a bug that in the function ngx_http_subrequest, it will make the …

fishgege opened the pull request:
…headers_in.headers  incorrect，and it will cause many problems

//sr->headers_in = r->headers_in; 
This line of code copy the headers  ,but it  maybe make the struct  mistake that  sr->headers_in.headers.last  mismatch &sr->headers_in.headers.part, when only one part be copied, the mismatch happen. if somewhere use  sr->headers_in.headers.last ，it will cause assert or some other problems。fx：in the function ngx_http_headers_more_assert 
so  only when there is only one part, the  mismatch happen  , and I update for this.  
By the way thanks for RaymondSchnyder ,  i got a mistake with code  in previous pull request, he find it

Fleshgrinder commented:
Sorry, I wanted to update my fork and not your master, the GitHub interface is sometimes confusing.

fishgege committed:
fix a bug that in the function ngx_http_subrequest, it will make the headers_in.headers  incorrect，and it will cause many problems

//sr->headers_in = r->headers_in; 
This line of code copy the headers  ,but it  maybe make the struct  mistake that  sr->headers_in.headers.last  mismatch &sr->headers_in.headers.part, when only one part be copied, the mismatch happen. if somewhere use  sr->headers_in.headers.last ，it will cause assert or some other problems。fx：in the function ngx_http_headers_more_assert 
so  only when there is only one part, the  mismatch happen  , and I update for this.  
By the way thanks for RaymondSchnyder ,  i got a mistake with code  in previous pull request, he find it

fishgege committed:
Update ngx_http_core_module.c

fishgege committed:
update

update for this pr

kacperduras commented:
```
Pull requests on GitHub cannot be accepted and will be automatically closed.
```

Look at the this repository header.
